### PR TITLE
feature: flag to allow to not close k8s client on stop

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java [hunk 1]
```diff
@@ -24,16 +24,24 @@
...
   private static final Logger log = LoggerFactory.getLogger(Operator.class);
   private final KubernetesClient kubernetesClient;
+  private final boolean closeKubernetesClientOnStop;
   private final ConfigurationService configurationService;
   private final ControllerManager controllers = new ControllerManager();
 
+
   public Operator(ConfigurationService configurationService) {
-    this(new DefaultKubernetesClient(), configurationService);
+    this(new DefaultKubernetesClient(), configurationService, true);
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:33]: Maybe the default should be `false`, no?

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:33]: Mentioned it in the issues (linked) IMHO usually (at least in our samples) it is used this way, that the registered hook shuts down the operator and also the client.

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:33]: Ok, we need to document it, I guess :)

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:33]: Actually, can we put this configuration option on the `ConfigurationService`? I think it makes sense to put it there and would avoid creating another constructor.

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:33]: good point, will change it a look how it would look like

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java]: updated

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java [hunk 2]
```diff
@@ -118,4 +118,8 @@ default Metrics getMetrics() {
   default ExecutorService getExecutorService() {
     return Executors.newFixedThreadPool(concurrentReconciliationThreads());
   }
+
+  default boolean getCloseClientOnStop() {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java:122]: How about just `closeClientOnStop`? I always find it weird to see `getDoSomething` methods :)

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java]: agree, changed

**Review by metacosm — APPROVED** [review]

**Pull request merged**